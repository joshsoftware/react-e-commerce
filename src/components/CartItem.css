.img_style {
    max-width: 80%;
    max-height: 200px !important;
    flex-shrink: 1 !important;
}

.img_style_card {
    max-width: 80%;
    height: 200px !important;
    flex-shrink: 1 !important;
}

.row_style {
    background-color: white;
    margin: 8px 0;
}

.col_one {
    text-align: center;
    flex-direction: column;
    margin: auto;
}

.col_two {
    margin: auto;
}

.col_five {
    text-align: center;
    margin: auto;
}
.cart-image {
    height: 250px;
}
.col_three {
    text-align: right;
    margin: auto;
    
}
.col_four {
    text-align: center;
    margin: auto;
}

.card_footer {
    padding-left: 20px !important;
}

.btn_cursor {
    cursor: not-allowed !important;
    pointer-events: all !important;
}

.update_profile {
    margin-top: 120px;
}

.image_size {
    width: 200px;
    height: 200px;
}
.file_field {
    display: inline-flex !important;
    width: auto !important;
    margin-top: 5px;
}
