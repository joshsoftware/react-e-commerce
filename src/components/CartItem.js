import React, { useEffect, useState } from 'react';
import './CartItem.css';
import ColumnWrapper from './ColumnWrapper';
import RowWrapper from './RowWrapper';
import CardImgWrapper from './CardImgWrapper';
import 'bootstrap/dist/css/bootstrap.min.css';
import PropTypes from 'prop-types';
import ButtonWrapper from './ButtonWrapper';
import CardTitleWrapper from './CardTitleWrapper';
import CardTextWrapper from './CardTextWrapper';
import CartDropdownContainer from '../containers/CartDropdownContainer';
import { deleteCartItem } from '../actions/cartActions';
import { deleteCartItemApi } from '../apis/cartApi';
import { useSelector } from 'react-redux';
import { getProductByIdApi } from '../apis/productApi';

const CartItem = ({ item, dispatch }) => {
  let { product_title, image_url, product_price, quantity, id } = item;
  const { userDetails } = useSelector((state) => state.loginReducer);
  let column_content = [];
  let i = 0;
  let item_details = [];
  item_details.push(<CardTitleWrapper key={i++} title={product_title} />);
  item_details.push(<CardTextWrapper key={i++} text={'Price : $ ' + product_price} />);
  item_details.push(<CartDropdownContainer id={id} dispatch={dispatch} quantity={quantity} />);
  column_content.push(
    <ColumnWrapper
      key={i++}
      className={'col_one'}
      md={3}
      xs={3}
      sm={3}
      lg={3}
      xl={3}
      data={<CardImgWrapper src={image_url} className="img_fluid img_style" />}
    />
  );
  column_content.push(<ColumnWrapper key={i++} className={'col_two'} data={item_details} />);
  column_content.push(
    <ColumnWrapper
      key={i++}
      className={'col_three'}
      data={
        <ButtonWrapper
          buttonText={'X'}
          onClick={() => {
            deleteCartItemApi({ token: userDetails.token, product_id: id });
            dispatch(deleteCartItem(id));
          }}
        />
      }
    />
  );

  return (
    <>
      <RowWrapper className={'row_style'} data={column_content} /> <hr></hr>
    </>
  );
};

export default CartItem;

CartItem.propTypes = {
  item: PropTypes.shape({
    image_url: PropTypes.string,
    product_title: PropTypes.string,
    product_price: PropTypes.number,
    quantity: PropTypes.number,
    id: PropTypes.number
  }),
  dispatch: PropTypes.func.isRequired
};
