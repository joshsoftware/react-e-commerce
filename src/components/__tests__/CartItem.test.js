import React from 'react';
import CartItem from '../CartItem';
import { render } from '@testing-library/react';

describe('CartItem Component', () => {
  it('Must return cart item', () => {
    const { asFragment } = render(
      <CartItem item={{ product_price: 15, product_title: 'product' }} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
