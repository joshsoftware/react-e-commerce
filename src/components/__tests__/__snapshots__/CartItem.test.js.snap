// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Component Must return cart item 1`] = `
<DocumentFragment>
  <div
    class="row_style row"
  >
    <div
      class="col_one col-3 col-sm-3 col-md-3 col-lg-3 col-xl-3"
    >
      <img
        alt="Product Image"
        class="img_fluid img_style card-img"
        src="https://assets.ajio.com/medias/sys_master/root/hb7/h8b/15083561943070/-473Wx593H-460422382-black-MODEL.jpg"
      />
    </div>
    <div
      class="col_two col"
    >
      <div
        class="card-title"
      >
        product
      </div>
      <p
        class="card-text"
      >
        Price : $ 15
      </p>
      <p
        class="card-text"
      >
        Quantity : $ 15
      </p>
    </div>
    <div
      class="col_three col"
    >
      <button
        class="btn btn-secondary"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
